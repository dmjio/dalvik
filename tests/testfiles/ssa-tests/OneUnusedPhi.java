// The SSA transformation should eliminate some kinds of unused
// values.  In particular, it should not create phi nodes that are
// never used.
class OneUnusedPhi {
  static int target;
  static void entry(int x) {
    int i1;
    int i2;
    if(x > 0) {
      i1 = x;
      i2 = 12;
    }
    else {
      i1 = 0;
      i2 = 22;
    }

    target = i1;
  }
}
