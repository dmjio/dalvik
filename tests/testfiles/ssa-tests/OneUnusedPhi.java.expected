SimplePhi { numBlocks = 3
          , phis = [ (0, fromList ["0", "%%arg0"])
                   ]
          , blockPhis = [(0, 1)]
          , assignedPhis = [(0, "target")]
          }